src/wp-includes/embed.php

```diff
@@ -1244,3 +1244,23 @@ function wp_filter_pre_oembed_result( $result, $url, $args ) {
 
 	return $result;
 }
+
+/**
+ * Applies the loading="lazy" attribute to the oEmbed iframe.
+ *
+ * @since 6.4.0
+ *
+ * @param string $html The oEmbed HTML, usually containing an iframe.
+ *
+ * @return string The oEmbed HTML with the loading="lazy" attribute added to the iframe.
+ */
+function wp_lazy_load_oembed_iframe( $html ) {
+	if ( $html && wp_lazy_loading_enabled( 'iframe', 'oembed' ) ) {
+		$tag_parser = new WP_HTML_Tag_Processor( $html );
+		if ( $tag_parser->next_tag( 'iframe' ) && ! $tag_parser->get_attribute( 'loading' ) ) {
+			$html = wp_iframe_tag_add_loading_attr( $html, 'oembed' );
```

Comment:
As noted on the ticket, let's try to figure out why post embeds fail when lazy-loading. If we can, then we can eliminate the following code from `wp_iframe_tag_add_loading_attr()`:

https://github.com/WordPress/wordpress-develop/blob/3dff98beb4384f0ffb13961c71178dedccf480a0/src/wp-includes/media.php#L2173-L2179

I think actually that I may have fixed this issue with https://github.com/WordPress/wordpress-develop/commit/02a2f9c9f515eee1a10c4fa04d95b89c1a750caf since in that commit it:

> Send a `ready` message from the host page to each post embed window in case the `iframe` loads before the `wp-embed` script does. When the `ready` message is received by the post embed window, it sends the same `height` message as it sends when it loads.

This landed in 5.9. But maybe this doesn't solve the problem since the issue is actually the inverse?

Reply:
I'll take a look to see if I can tell why this happens and how to fix it.